src/ui/windows/TogglDesktop/TogglDesktop/ui/views/Timeline.xaml.cs

```diff
@@ -35,15 +42,29 @@ protected override void OnPropertyChanged(DependencyPropertyChangedEventArgs e)
                 _disposable = new CompositeDisposable();
 
                 ViewModel?.WhenAnyValue(x => x.SelectedScaleMode).Buffer(2, 1)
-                    .Select(b => (double)ViewModel.ScaleModes[b[1]] / ViewModel.ScaleModes[b[0]])
-                    .Subscribe(ratio => MainViewScroll.ScrollToVerticalOffset(MainViewScroll.VerticalOffset * ratio))
+                    .Select(b => (double)TimelineViewModel.ScaleModes[b[1]] / TimelineViewModel.ScaleModes[b[0]])
+                    .Subscribe(ratio => SetMainViewScrollOffset(MainViewScroll.VerticalOffset * ratio))
                     .DisposeWith(_disposable);
-                ViewModel?.WhenAnyValue(x => x.SelectedDate).Where(date => date.Date == DateTime.Today)
-                    .Subscribe(_ => SetScrollToCurrentTime())
+                ViewModel?.WhenValueChanged(x => x.IsTodaySelected)
+                    .Where(x => x)
+                    .ObserveOn(RxApp.MainThreadScheduler)
+                    .Subscribe(_ => MainViewScroll.ScrollToVerticalOffset(ViewModel.CurrentTimeOffset - MainViewScroll.ActualHeight / 2))
+                    .DisposeWith(_disposable);
+                ViewModel?.WhenAnyValue(x => x.FirstTimeEntryOffset)
+                    .Where(_ => !ViewModel.IsTodaySelected)

```

Comment:
Would be nice to use `.ObserveOn(RxApp.MainThreadScheduler)` also here and get rid of overcomplicated ViewModel getter.

Reply:
53516b1 looks good 👍